net/can/can_callback.c

```diff
@@ -236,6 +234,7 @@ uint16_t can_datahandler(FAR struct net_driver_s *dev,
   else
     {
       nerr("ERROR: Failed to queue the I/O buffer chain: %d\n", ret);
+      ret = 0;
```

Comment:
why zero ret

Reply:
This was actually a bug. This function returns an uint16_t as the number of bytes actually buffered, but the call to iob_tryadd_queue (line 218)  returns an int and, if it fails, it returns a negative value. In that case, can_datahandler return value is cast to an unsigned value as if it would had buffered many bytes. 
Setting the returning value to zero when iob_tryadd_queue returns a negative value reflects the actual meaning that no bytes have been buffered

Reply:
it's better to move this change to a new patch and give a clear commit message

Reply:
I think this should be part of this PR, otherwise the CAN statistics implemented in this patch would not work properly. I can create a specific commit for this patch if you prefer

Reply:
done @xiaoxiang781216